### ISSUE TITLE
JS Backend: v0.5.16: Wrong generation of static native fields

### ISSUE THREAD

**Issue by marciodel** [jtransc/jtransc#125]
Hi, I'm trying to create a Typed DOM implementation. When I try to declare a static field the compiler generates wrong code:

```
@JTranscNativeName("window")
public class Global {
	public static Window window;
}

@JTranscNativeName("Window")
public abstract class JsWindow {
	public Console console;
	public String name;

	public native void alert(String s);

	public void alert(int s) {
	}
}

@JTranscNativeName("Console")
public abstract class Console {
	native void log(Object... args);
}

public class Main {

	public static void main(String[] args) {
		Global.window.alert(2);
		Global.window.console.log("1");
       }

}


```

It generates the following code 

```
test_Main["main([Ljava/lang/String;)V"] = function(p0) { 
	var fA1 = null, tA0 = null, fA0 = null;
	test_Global.window.alert(2);
	fA0 = (test_Global.window.console);
	tA0 = (new JA_L(1, "[Ljava.lang.Object;"));
	fA1 = tA0;
	(tA0).data[0] = (S[190]);
	(fA0).log(N.unbox((fA1)));
	return;
};

```
The problem is that test_Global is not declared because it is a native class.

**Comment by soywiz** [comment]
Going to check. Are you using git version? Or lastest published version? If it fails, I will fix it asap.

**Comment by marciodel** [comment]
I'm using the latest published version.
Thanks!

**Comment by soywiz** [comment]
Fixed already. It will be included in 0.6.0, which hopefully will be released later today (depending on the timezone).

It require some adjustments though:
https://github.com/jtransc/jtransc/blob/2ef042cbcd69fb7f236f0d5e6da8f826095b1396/jtransc-main/test/jtransc/jtransc/nativ/JTranscJsNativeMixedTest.java

You cannot use variadic calls yet. So for now, you will have to add some overloads and annotate arguments with `@JTranscUnboxParam` or it will display `java_lang_Integer {1}` instead.

This works in node.js:
```
static public void main(String[] args) {
		Global.global.console.log(1);
		Global.global.console.log("hello");
}

@JTranscNativeName("global")
public abstract static class JsGlobal {
	public Console console;
	public String name;

	//public native void alert(String s);
	//public native void alert(int s);
}

@JTranscNativeName("global")
public static class Global {
	public static JsGlobal global;
}

@JTranscNativeName("Console")
public abstract static class Console {
	native public void log(@JTranscUnboxParam  Object a);
}
```

Also in 0.6.0 for Java (and kotlin has better wrappers in `jtransc-rt-core-kotlin`), you can use JsDynamic. That allows you to dynamically call JS without static typings like this. In java would be like this:

```
int res = JsDynamic.global("console").get("log").toInt();
System.out.println(res);
System.out.println(JsDynamic.global("Math").call("max", -4, -3).toInt() == -3);
System.out.println(JsDynamic.global().get("Date").newInstance(1234567).call("getTime").toInt());
```

**Comment by soywiz** [comment]
Also since JsDynamic uses `@JTranscCallSiteBody`, generated code will be like plain JS when using JsDynamic:

```
lI1 = ((_global["console"]["log"])|0);
java_lang_System._out["println(I)V"](lI1);
fA0 = java_lang_System._out;
fI1 = ((((((_global["Math"]["max"](-4, -3))|0) == -3))) ? (1) : (0));
fA0["println(Z)V"](((fI1)!=0));
```

Take into account that in JS doing `_global["Math"]` is equivalent to `_global.Math` and won't have any performance penalties.
